palindromic-substrings/whewchews.ts

```diff
@@ -0,0 +1,35 @@
+function countSubstrings(s: string): number {
+  // SC: O(N^2)
+  const dict: Map<string, boolean> = new Map();
+  const n = s.length;
+
+  // TC: O(N^2)
+  for (let start = 0; start < n; start++) {
+    for (let end = start; end >= 0; end--) {
```

Comment:
`start`는 감소시키고, `end`는 증가하면서 이중 루프를 돌아야하지 않을까요?
현재 코드는 루프 방향이 정반대로 되어 있어서 리트코드에 제출하시면 오답이 나올 것 같습니다.

```suggestion
  for (let end = 0; end < n; end++) {
    for (let start = end; start >= 0; start--) {
```

```suggestion
  for (let start = n; start >= 0; start--) {
    for (let end = start; end < n; end++) {
```

Reply:
감사합니다! 리뷰해주신대로 이중루프를 돌 때 더 짧은 문자열을 먼저 처리하고, 이를 기반으로 더 긴 부분 문자열을 처리해주어야 제대로 작동하겠네요. 리뷰 반영했습니다!